### DISCUSSION TITLE
Nao valida o Hash e tax exemption reason incorrecto

### DISCUSSION THREAD

**Discussion by FernandoManuel6107** [assoft-portugal/SAF-T-AO#181] [Q&A] [answered]
Saudações ilustres,

Ao testar o saft em anexo com apenas 3 facturas obtenho as seguintes mensagens

Erro no ficheiro,AuditFile.SourceDocuments.SalesInvoices.2Invoice (FT 2021/2) . Hash está errado.
Erro no ficheiro,AuditFile.SourceDocuments.SalesInvoices.3Invoice (FT 2021/3) . Hash está errado.
Erro no ficheiro, AuditFile.SourceDocuments.SalesInvoices.1Invoice.1Line.TaxExemptionReason (FT 2021/1) está mal preenchido.
Erro no ficheiro, AuditFile.SourceDocuments.SalesInvoices.3Invoice.1Line.TaxExemptionReason (FT 2021/3) está mal preenchido.

Me ajudem por favor
[SAFT.txt](https://github.com/assoft-portugal/SAF-T-AO/files/6864735/SAFT.txt)

**Comment by cryptolopes** [comment, marked as answer]
Noutros [Issues](https://github.com/assoft-portugal/SAF-T-AO/issues) e [Discussions](https://github.com/assoft-portugal/SAF-T-AO/discussions) existe já muita informação de como ultrapassar os problemas as assinaturas.
